src/SkyApm.Agent.Hosting/Extensions/ServiceCollectionExtensions.cs

```diff
@@ -46,10 +47,30 @@ public static class ServiceCollectionExtensions
     {
         public static IServiceCollection AddSkyAPM(this IServiceCollection services, Action<SkyApmExtensions> extensionsSetup = null)
         {
+            string environment = Environment.GetEnvironmentVariable("ASPNETCORE_ENVIRONMENT");
+            if (environment == null || environment.Length < 1)
+            {
+                environment = "Development";
+            }
+            IConfigurationBuilder configurationBuilder = new ConfigurationBuilder();
+            configurationBuilder.AddJsonFile("skyapm.json", true);
+            configurationBuilder.AddJsonFile("skyapm." + environment + ".json", true);
+            IConfiguration configuration = configurationBuilder.Build();
+            string enable = configuration?.GetSection("SkyWalking:Enable").Value ?? "false";
+            if (null == enable || !"true".Equals(enable.ToLower()))
+            {
+                return services;
+            }
+            string serviceName = configuration?.GetSection("SkyWalking:ServiceName").Value ?? "";

```

Comment:
get service name from config file instead of environment variable (for windows services)